server/resources/resourcesRepository.js

```diff
@@ -49,8 +69,27 @@ export const count = async ({ draft }) => {
 	return parseInt(count, 10);
 };
 
-export const findAll = async ({ draft, limit, offset }) => {
-	const { rows } = await db.query(pagedResourceQuery, [draft, limit, offset]);
+export const findAll = async ({
+	id,
+	is_admin,
+	draft,
+	account,
+	limit,
+	offset,
+}) => {
+	let query;
+	let param = [limit, offset];
+	if (is_admin && account) {
+		query = pagedResourceAdminQuery;
+		param.unshift(id);
+	} else if (!is_admin && account) {
+		query = pagedResourceUserQuery;
+		param.unshift(id);
```

Comment:
💡 The first two cases are basically the same. This would be clearer in my opinion if `account` was _first_, but another way to do it is collapse them entirely:
```suggestion
	if (account) {
		query = is_admin ? pagedResourceAdminQuery : pagedResourceUserQuery;
		param.unshift(id);
```